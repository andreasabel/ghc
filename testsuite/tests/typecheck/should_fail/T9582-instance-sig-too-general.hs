-- 2014-09-18 Andreas Abel
-- Failing test case for ticket 9582

{-# LANGUAGE InstanceSigs, TypeFamilies #-}

module Fail where

class C a where
  type T a
  m :: T a

instance C Int where
  type T Int = String

  -- The following type signature for m is currently rejected,
  -- as it is too general.
  m :: Show a => [a]
  m = []

  -- However, in principle we could allow method instances of
  -- a more general type than expected, and insert a coercion
  -- back to the more specialized, expected type.
