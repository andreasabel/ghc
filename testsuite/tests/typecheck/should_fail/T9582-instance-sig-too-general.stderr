
T9582-instance-sig-too-general.hs:17:8:
    Couldn't match type ‘[Char]’ with ‘forall a. Show a => [a]’
    Expected type: T Int
      Actual type: forall a. Show a => [a]
    In the instance declaration for ‘C Int’
